src/macaron/errors.py

```diff
@@ -58,3 +58,15 @@ class InvalidHTTPResponseError(MacaronError):
 
 class CheckRegistryError(MacaronError):
     """The Check Registry Error class."""
+
+
+class ParseError(MacaronError):
+    """The errors related to parsers."""
+
+
+class CallGraphError(MacaronError):
```

Comment:
Yes, it is intended to be used for all parsers. In general, we shouldn't create new error classes if not really necessary, so this error class is made to be generic on purpose to cover all parsers.

>should we make CallGraphError inherits from ParseError
I don't think so, because not all `CallGraphError` errors are related to `ParseError` errors.

Reply:
Right I see. Thanks @behnazh-w .
> Yes, it is intended to be used for all parsers.

Then should we make the `parse` function of the `bashparser.py` module raises ParseError too ?

Reply:
See commit f0d89a961fe4380ceeb89526ff1e7cbacd680294